### ISSUE TITLE
Alert ignores environment setting

### ISSUE THREAD

**Issue by jannikbertram** [getsentry/sentry#58845]
### Environment

SaaS (https://sentry.io/)

### Steps to Reproduce

1. Create an alert for a rule with targeted to environment `prod`
2. Configure an action to send those alerts to a Slack channel
3. Trigger an alert with environment `dev`

### Expected Result

The alert does not show up in the Slack channel

### Actual Result

The alert does show up in the Slack channel. I'm not sure if it happens always or randomly. We have been seeing this issue many many times in our team, but not across all projects and alerts.

### Product Area

Alerts

### DSN

https://143eadef6c7f4610967e80fba5fbe7f2@o468865.ingest.sentry.io/5497530

**Comment by roggenkemper** [comment]
Do you mind sending me a link to the alert so i can take a look ? If you don't want to post it on Github, feel free to email it to me at richard.roggenkemper@sentry.io

**Comment by jannikbertram** [comment]
Hey @roggenkemper, while trying to find the alert I was talking about, I made an interesting observation. First of all, [here is the link to the alert](https://allgravy.sentry.io/issues/4521897071/?alert_rule_id=2551081&alert_type=issue&notification_uuid=6cdc2a08-3438-4930-a452-cd56381a88ec&project=5497530&referrer=slack)

I can see on Slack that we were discussing with the team why this alert originates from environment `localhost` and still shows up in our production alert channel. Now, as I'm clicking the link again, it does show an error from environment `production`. I suspect that the link now leads to a different error than it did right after it was posted to Slack.

[This one](https://allgravy.sentry.io/issues/4283132458/?referrer=slack&notification_uuid=786c40de-2b8e-447e-8938-9feb71395a9f&alert_rule_id=14526206&alert_type=issue) is another example where I'm sure it showed an error from environment `development` when I clicked it the first time.

The two issues have in common that they mostly occur in non-production environments (99% and 90% respectively). 

Is there a chance that the link posted by an alert does not guarantee to show an error of the same environment as the alert?

**Comment by scefali** [comment]
@jannikbertram What's the actual alert? Is it [this](https://allgravy.sentry.io/alerts/rules/api/2551081/)? Cause I don't see anything about only alerting for the prod env.

**Comment by jannikbertram** [comment]
@scefali yep it's that. It's configured only for the `production` environment
![image](https://github.com/getsentry/sentry/assets/17082165/7ea09b92-9fb0-43a6-97a6-40793cff45e4)

Here is another example from today.
[Sentry issue](https://allgravy.sentry.io/issues/4283132458/?alert_rule_id=14526206&alert_type=issue&notification_uuid=cb8e4e30-f273-49ee-abcb-9ee57e148f28&project=6681138&referrer=slack)
[Alert rule](https://allgravy.sentry.io/alerts/rules/org-admin/12510095/)


To be a bit clearer here:
It seems like the alert in our Slack channel points to an issue which occurs both in `development`, but also `production`. That means this issue contains events from both environments.

**Expected behaviour**
Since our alert is only configured for production, we expect to see only production events. When we check the alerts, we don't care about the development events. Mixing them actually makes debugging much harder for us.

**Actual behaviour**
After clicking the alert, we see an event from the `development` environment with host `http://localhost:3000`. When clicking on "All Events", we can see both events from `development` and `production`

**Comment by ceorourke** [comment]
@jannikbertram are you saying that it's actually firing on development environment events, or that it's firing on production env events but it's confusing to sift through both? Is it possible it's firing appropriately only on production env events but a development env event has come in by the time you clicked on it? Have you used the environment selector on the issue details page to make looking through all events easier?

**Comment by jannikbertram** [comment]
Yeah it's most likely as you say @ceorourke. I haven't compared the timestamps yet, but it would explain why we this occurs randomly for us.

Even if this is not a bug, I want to point out that our whole team perceives this as a bug. It actually led to us not looking into some critical errors because we assumed they only occured in the development environment, but they did in fact also appear in production.

It would be nice if production alerts would contain a query parameter that would automatically filter the Events for that environment. Maybe we can turn this into a feature request?

**Comment by ceorourke** [comment]
@jannikbertram I like that solution - just to clarify it's only when the alert rule filters by the environment that the alert link would have the environment query parameter, right?

**Comment by jannikbertram** [comment]
> @jannikbertram I like that solution - just to clarify it's only when the alert rule filters by the environment that the alert link would have the environment query parameter, right?

Exactly, if you select all environments, it should work like it does today

**Comment by ceorourke** [comment]
Hi @jannikbertram I merged a PR to address this: https://github.com/getsentry/sentry/pull/59949 lmk if you have any issues with it!

**Comment by jannikbertram** [comment]
Fantastic @ceorourke I can already see that the environment parameter is added to the links.
Thanks for the quick turnaround time! 🙌
